src/yb/rocksdb/util/thread_posix.cc

```diff
@@ -238,4 +238,15 @@ int ThreadPool::UnSchedule(void* arg) {
   return count;
 }
 
+std::vector<std::string> ThreadPool::GetThreadpoolWaitStates() {
+  std::vector<std::string> res;
+  PthreadCall("lock", pthread_mutex_lock(&mu_));
```

Comment:
What is this lock/unlock protecting?

Reply:
This lock/unlock should protect `bg_wait_states_` similar to how they protect `bgthreads_`. I added these when the updates were also being protected by the same lock, but now the updates are not protected (in this PR), I think it's safe to be removed.

Reply:
This lock also protects bg_wait_states_ being inserted into or removed from.